server/pulp/server/webservices/views/roles.py

```diff
@@ -0,0 +1,229 @@
+import json
+
+from django.core.urlresolvers import reverse
+from django.utils.encoding import iri_to_uri
+from django.views.generic import View
+from django.http import HttpResponseRedirect
+from pulp.server import exceptions as pulp_exceptions
+from pulp.server.auth import authorization
+from pulp.server.managers import factory
+from pulp.server.webservices.controllers.base import json_encoder as pulp_json_encoder
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices.views.util import (generate_json_response,
+                                                generate_json_response_with_pulp_encoder)
+
+
+class RolesView(View):
+    """
+    Views for roles.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request):
+        """
+        List all roles.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :return: Response containing a list of roles
+        :rtype: django.http.HttpResponse
+        """
+        role_query_manager = factory.role_query_manager()
+        user_query_manager = factory.user_query_manager()
+        permissions_manager = factory.permission_manager()
+        roles = role_query_manager.find_all()
+        for role in roles:
+            role['users'] = [u['login'] for u in
+                             user_query_manager.find_users_belonging_to_role(role['id'])]
+
+            resource_permission = {}
+            # isolate schema change
+            if role['permissions']:
```

Comment:
This is completely correct and fine, but just so you're aware of another stylistic option, some people would write these two lines as just:

``for item in role['permissions'] or []:``

which saves a level of indentation and only accesses the value in ``role`` once. But it's really just a stylistic preference, and you're welcome to keep what you have here if you like. I personally don't feel strongly either way.